variables/yaml_helpers.go

```diff
@@ -399,10 +399,38 @@ func unmarshalBooleanField(fields map[string]interface{}, fieldName string, requ
 	}
 }
 
+// parseVariablesFromEnvironmentVariables parses variables from environment variables
+//
+// These variables are expected to be in the format:
+//
+//	BOILERPLATE_var_name=value
+func parseVariablesFromEnvironmentVariables() (map[string]any, error) {
+	vars := map[string]any{}
+
+	for _, envVar := range os.Environ() {
+		key, value, found := strings.Cut(envVar, "=")
+		if !found {
+			return vars, errors.WithStackTrace(InvalidVarSyntax(envVar))
+		}
+
+		if strings.HasPrefix(key, "BOILERPLATE_") {
```

Comment:
Move this guard before the `strings.Cut`, we don't want to exit if the environment has a weird env var that doesn't contain `=` somehow and isn't related to boilerplate.